File: src/main/java/com/typelead/intellij/plugin/eta/lang/EtaLanguage.java
package com.typelead.intellij.plugin.eta.lang;

import com.intellij.lang.Language;

public class EtaLanguage extends Language {

    public static final EtaLanguage INSTANCE = new EtaLanguage();

    private EtaLanguage() {
        super("Eta");
    }
}

File: src/main/eta/FFI/Com/TypeLead/IntelliJ/Plugin/Eta/Lang/EtaLanguage.hs
module FFI.Com.TypeLead.IntelliJ.Plugin.Eta.Lang.EtaLanguage where

import P
import FFI.Com.IntelliJ.Lang.Language

data {-# CLASS "com.typelead.intellij.plugin.eta.lang.EtaLanguage" #-}
  EtaLanguage = EtaLanguage (Object# EtaLanguage)
  deriving Class

type instance Inherits EtaLanguage = '[Object, Language]

foreign import java unsafe
  "@static com.typelead.intellij.plugin.eta.lang.EtaLanguage.INSTANCE"
  etaLanguage :: EtaLanguage
